### PR TITLE
python310Packages.pynndescent: 0.5.8 -> 0.5.10

### PR DESCRIPTION
**Pull request by natsukium**
## Description of changes

Diff: https://github.com/lmcinnes/pynndescent/compare/0.5.8...0.5.10

Changelog: https://github.com/lmcinnes/pynndescent/releases/tag/0.5.10

- add patch for scipy 1.11

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pynndescent/default.nix [hunk 1]
```diff
@@ -14,27 +13,21 @@
...
-      hash = "sha256-mbe01BwroS5q6hENsj3NejmGGhmk2IeX4LD6Iq6PR0c=";
-    })
-    (fetchpatch {
-      url = "https://github.com/lmcinnes/pynndescent/commit/e56b92776a4a05f2dabb80d25479bd37e7ebd88e.patch";
-      hash = "sha256-zVTaW4syGEHh2HAGPyBN3YXqUGe55v/LxKLX/zjXT5Y=";
-    })
-  ];
+  postPatch = ''
+    # fix scipy 1.11 compat
+    substituteInPlace pynndescent/tests/test_distances.py \
```
Comment (Mic92) [inline pkgs/development/python-modules/pynndescent/default.nix:28]: Can you fetch this patch instead? https://github.com/lmcinnes/pynndescent/commit/86e0d716a3a4d5f4e6a0a3c2952f6fe339524e96.patch

https://github.com/lmcinnes/pynndescent/pull/224

Comment (natsukium) [reply pkgs/development/python-modules/pynndescent/default.nix]: done.
Thanks for your review!

**Comment by tjni** [comment]
Result of `nixpkgs-review pr 248481` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.scikit-tda</li>
    <li>python310Packages.scikit-tda.dist</li>
    <li>python310Packages.umap-learn</li>
    <li>python310Packages.umap-learn.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.pynndescent</li>
    <li>python310Packages.pynndescent.dist</li>
  </ul>
</details>

**Comment by tjni** [comment]
Result of `nixpkgs-review pr 248481` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.scikit-tda</li>
    <li>python310Packages.scikit-tda.dist</li>
    <li>python310Packages.umap-learn</li>
    <li>python310Packages.umap-learn.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.pynndescent</li>
    <li>python310Packages.pynndescent.dist</li>
  </ul>
</details>

**Comment by tjni** [comment]
I can reproduce the segmentation fault on my aarch64-linux VM. I'm not sure what's going on.

**Comment by natsukium** [comment]
I suspect numba is the cause.
Another case is sparse, which causes a similar segmentation fault.
https://hydra.nixos.org/build/232432752

**Comment by tjni** [comment]
Yes, it's something related to that. Here's a backtrace without any debugging symbols (haven't figured out how to do that) or any sophistication:

```
pynndescent-aarch64-linux> pynndescent/tests/test_distances.py ......................
pynndescent-aarch64-linux> Program received signal SIGSEGV, Segmentation fault.
pynndescent-aarch64-linux> 0x00000508c85f7c08 in ?? ()
pynndescent-aarch64-linux> --Type <RET> for more, q to quit, c to continue without paging--#0  0x00000508c85f7c08 in ?? ()
pynndescent-aarch64-linux> #1  0x0000ffffe58a4424 in NRT_MemInfo_call_dtor ()
pynndescent-aarch64-linux>    from /nix/store/qj7qj2c431jb7xfwiz8gl6p59pvyw0m6-python3.10-numba-unstable-2023-08-02/lib/python3.10/site-packages/numba/core/runtime/_nrt_python.cpython-310-aarch64-linux-gnu.so
pynndescent-aarch64-linux> #2  0x0000fffff71aa660 in pynndescent::sparse::sparse_canberra[abi:v107][abi:c8tJTIcFKzyF2ILShI4CrgQElQb6HczSBAA_3d](Array<int, 1, C, mutable, aligned>, Array<float, 1, C, mutable, aligned>, Array<int, 1, C, mutable, aligned>, Array<float, 1, C, mutable, aligned>) ()
pynndescent-aarch64-linux> #3  0x0000fffff71aaa70 in cpython::pynndescent::sparse::sparse_canberra[abi:v107][abi:c8tJTIcFKzyF2ILShI4CrgQElQb6HczSBAA_3d](Array<int, 1, C, mutable, aligned>, Array<float, 1, C, mutable, aligned>, Array<int, 1, C, mutable, aligned>, Array<float, 1, C, mutable, aligned>) ()
pynndescent-aarch64-linux> #4  0x0000ffffe5c043e0 in call_cfunc(Dispatcher*, _object*, _object*, _object*, _object*) ()
pynndescent-aarch64-linux>    from /nix/store/qj7qj2c431jb7xfwiz8gl6p59pvyw0m6-python3.10-numba-unstable-2023-08-02/lib/python3.10/site-packages/numba/_dispatcher.cpython-310-aarch64-linux-gnu.so
pynndescent-aarch64-linux> #5  0x0000ffffe5c047a8 in compile_and_invoke(Dispatcher*, _object*, _object*, _object*) ()
pynndescent-aarch64-linux>    from /nix/store/qj7qj2c431jb7xfwiz8gl6p59pvyw0m6-python3.10-numba-unstable-2023-08-02/lib/python3.10/site-packages/numba/_dispatcher.cpython-310-aarch64-linux-gnu.so
pynndescent-aarch64-linux> --Type <RET> for more, q to quit, c to continue without paging--#6  0x0000ffffe5c04a90 in Dispatcher_call(Dispatcher*, _object*, _object*) ()
pynndescent-aarch64-linux>    from /nix/store/qj7qj2c431jb7xfwiz8gl6p59pvyw0m6-python3.10-numba-unstable-2023-08-02/lib/python3.10/site-packages/numba/_dispatcher.cpython-310-aarch64-linux-gnu.so
pynndescent-aarch64-linux> #7  0x0000fffff7c54ad8 in _PyObject_MakeTpCall ()
pynndescent-aarch64-linux>    from /nix/store/w9n23r5v993f26iawpqzm7cn200k0zdg-python3-3.10.12/lib/libpython3.10.so.1.0
pynndescent-aarch64-linux> #8  0x0000fffff7bfeac0 in call_function ()
pynndescent-aarch64-linux>    from /nix/store/w9n23r5v993f26iawpqzm7cn200k0zdg-python3-3.10.12/lib/libpython3.10.so.1.0
pynndescent-aarch64-linux> #9  0x0000fffff7c01d54 in _PyEval_EvalFrameDefault ()
pynndescent-aarch64-linux>    from /nix/store/w9n23r5v993f26iawpqzm7cn200k0zdg-python3-3.10.12/lib/libpython3.10.so.1.0
pynndescent-aarch64-linux> #10 0x0000fffff7da61a4 in _PyEval_Vector ()
```

**Comment by natsukium** [comment]
It is already included in staging-next.

**Pull request closed without merging**